Plant -> Harvest -> Store ->Transfer to a building where required

Naman-sopho opened the pull request:
Adds a content loop for the `Plant` task. After a Plant task is completed and the placed crops have grown fully a `Harvest` task is added to the Holding. After performing the Harvest task, these crops are removed from the farm and a `PlaceBlocksInStorage` task is added to the Holding.
#### Outstanding before merge
- [x] Implement `PlaceBlocksInStorage` task logic and behavior.
- [x] Complete the loop with tasks like taking blocks from storage and placing them in Diner for use.

Naman-sopho committed:
Add subsequentTask and delay field to Task class.

Implement new Harvest task type.

Naman-sopho committed:
Add Harvest as a subsequent task after Plant

Naman-sopho committed:
Implement Harvest task behavior

Naman-sopho committed:
Implement Harvest task logic.

Add new PlaceBlocksInStorage task type.

Cervator commented on src/main/java/org/terasology/Constants.java:
```diff
@@ -86,6 +86,7 @@
     public static final int PEDESTAL_REGION_INDEX = 2;
     public static final int LABORATORY_SLAB_REGION = 16;
 
+    public static final String AIR_BLOCK_URI = "engine:air";
```
If you can use a `BlockUri` instead of this String there's a constant available in `BlockManager.AIR_ID`

Also, perhaps rename this class to `MooConstants`? To avoid confusion / class name classes. Plus `MooConstants` sounds funny

Cervator commented on src/main/java/org/terasology/taskSystem/actions/AddCropTransferToStorageTaskNode.java:
```diff
@@ -0,0 +1,155 @@
+
+    @Override
+    public BehaviorState modify(Actor oreon, BehaviorState result) {
+        TaskComponent oreonTaskComponent = oreon.getComponent(TaskComponent.class);
```
This is a somewhat long'ish method with unclear grouping / newlines - maybe group into clear blocks and add a comment to each explaining the goal?

Cervator commented on src/main/java/org/terasology/taskSystem/tasks/PlaceBlocksInStorageTask.java:
```diff
@@ -0,0 +1,47 @@
+import org.terasology.taskSystem.AssignedTaskType;
+import org.terasology.taskSystem.Task;
+
+public class PlaceBlocksInStorageTask extends Task {
```
Maybe this and similar tasks could use a brief little javadoc comment outlining what the task is for (the name makes it fairly clear but maybe an example / some context could help? Like it is meant for when a task at a site has completed and there's output to move to storage)

Cervator reviewed (changes requested):
Reviewed early, minor comments nothing major. Waiting for some more implementation before testing :-)

Naman-sopho committed:
Implement block transfer to Storage chest after harvesting

Naman-sopho committed:
Implement block rendering for a task being performed

Naman-sopho committed:
Add javadoc and comments where required.

Change PlaceBlocksInStorage to a more generic PlaceBlocksInChest.

Naman-sopho committed:
Exclamation mark and book are now removed from the Pedestal after Research task

Naman-sopho committed:
Add resource transfer task from Storage to a building where it is required.

Subsequent tasks can now have a different region than the previous task.

Naman-sopho committed:
Implement new type handler after changes to the Task class.

Modify Task and TaskComponent for an easier TypeHandler implementation and modify other related systems to go with this.

mkienenb commented on src/main/java/org/terasology/research/ResearchSystem.java:
```diff
@@ -240,10 +245,32 @@ private void getResearchRecipe(EntityRef item, EntityRef inventoryEntity) {
+
+        for (EntityRef item : pedestalInventory.itemSlots) {
+            EntityRef removedItem = inventoryManager.removeItem(pedestalEntity, pedestalEntity, item, false);
+            logger.info(removedItem + "");
```
What's the + "" for?   If you need removedItem to be a string, use either .toString() or probably better String.valueOf() instead of +""

mkienenb reviewed (approved):
One minor improvement is all I saw.
